src/otx/v2/adapters/datumaro/adapter/detection_dataset_adapter.py

```diff
@@ -24,9 +24,8 @@ class DetectionDatasetAdapter(DatumaroDatasetAdapter):
 
     def get_otx_dataset(self) -> DatasetEntity:
         """Convert DatumaroDataset to DatasetEntity for Detection."""
-        # Prepare label information
-        label_information = self._prepare_label_information(self.dataset)
-        self.label_entities = label_information["label_entities"]
+        if not hasattr(self, "label_entities"):
+            super().get_label_schema()
```

Comment:
Calling get_somthing() but not actually getting something, which mean it's just being called for the internal side effect. I don't think this is quite readable code.

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2631/commits/3ca5154cc9fd8172865590ad46d03db01b772a91